kythe/go/serving/tools/write_tables/write_tables.go

```diff
@@ -61,6 +63,7 @@ var (
 	verbose = flag.Bool("verbose", false, "Whether to emit extra, and possibly excessive, log messages")
 
 	experimentalBeamPipeline = flag.Bool("experimental_beam_pipeline", false, "Whether to use the Beam experimental pipeline implementation")
+	beamShards               = flag.Int("beam_shards", 128, "Number of shards for beam processing.")
```

Comment:
Please set the default to `0` and state that a "reasonable" default will be chosen based on the input if given a non-positive value.  Then check for non-positive numbers below and set it to 128 otherwise,.